### PR TITLE
Avoid double installation of header files with TriBITS

### PR DESCRIPTION
**Pull request by jjwilke**
Addresses trilinos issue: https://github.com/trilinos/Trilinos/issues/7907

Hotfix in trilinos here: https://github.com/trilinos/Trilinos/pull/7920

### CODE REVIEW

File: cmake/kokkos_tribits.cmake [hunk 1]
```diff
@@ -361,10 +361,11 @@ FUNCTION(KOKKOS_ADD_LIBRARY LIBRARY_NAME)
   CMAKE_PARSE_ARGUMENTS(PARSE
     "ADD_BUILD_OPTIONS"
     ""
-    ""
+    "HEADERS"
     ${ARGN}
   )
   IF (KOKKOS_HAS_TRILINOS)
+    # Pull out the headers, we will manually install them
```
Comment (dalg24) [inline cmake/kokkos_tribits.cmake:368]: Where are they manually installed?

**Comment by crtrott** [comment]
Also need to go into the release candidate for 3.2.1

**Comment by jjwilke** [comment]
In Trilinos headers that exist in subfolders like impl were getting installed twice: once in a directory-preserving way to `$prefix/include/impl` (manually by Kokkos) and once to the top-level `include` directory (implicitly via TriBITS). The double header install breaks some downstream packages. This no longer passes the list of headers to TriBITS and prevents the second install.

**Comment by jjwilke** [comment]
This has no effect on standalone kokkos

**Comment by dalg24** [comment]
(Duplicating comment since I cannot open an issue from a review comment)

Make sure you cherry-pick these changes and open a PR against `release-candidate-3.2.1`

**Comment by ndellingwood** [comment]
PR #3320 with cherry-pick to release-candidate-3.2.1

**Pull request merged**